### ISSUE TITLE
Add support for xp

### ISSUE THREAD

**Issue by Felixmaker** [lc-soft/LCUI#234]
would it be possible to support xp?

**Comment by lc-soft** [comment]
This is possible if [VCPkg](https://github.com/microsoft/vcpkg) supports compilation of dependent libraries for Windows XP. Otherwise, you need to manually use the XP compilation toolchain to compile LCUI and its dependent libraries.

**Comment by Felixmaker** [comment]
I tried to build LCUI using /MT option, but it failed. Link error 2001 @_BrotliDecoderDecompress (freetype).

**Comment by lc-soft** [comment]
FreeType depends on Brotli, and you need to link to it.

**Comment by Felixmaker** [comment]
Thank you! I have built the dll, but when I replace the dll in my project, the program run failed, it does not show in screen and needs to kill in the task manager.

**Comment by lc-soft** [comment]
You didn't provide your build environment and build methods, so I couldn't give a solution.

Maybe you can run the program with debugging tools.

**Comment by lc-soft** [comment]
You didn't provide your build environment and build methods, so I couldn't give a solution.

Maybe you can run the program with debugging tools.

**Comment by Felixmaker** [comment]
Sorry, I use VS2019 and MSVC V142 (both are the latest version) to build LCUI.dll etc and use lcpkg to install all dependcies.

**Comment by lc-soft** [comment]
> I use VS2019 and MSVC V142 (both are the latest version) to build LCUI.dll etc

v142? Why not use v142_xp? You said you want to support windows XP, then you should use v142_xp.

In the solution configuration of LCUI, the platform toolset has selected v141_xp, you can switch to v142_xp, so you don't need to modify the platform.h file. 

![image](https://user-images.githubusercontent.com/1730073/119689848-c75dcc00-be7b-11eb-80f1-6564bb399548.png)

> the program run failed, it does not show in screen and needs to kill in the task manager.
> ... Then I built the program, but when I replace it in my project, it doesn't work.

Are you using VS2019 to build your project? If so, please use vs2019 to debug it. The general steps are as follows: 

1. Add a breakpoint on the line of code below `LCUI_Init()`.
    ![image](https://user-images.githubusercontent.com/1730073/119689780-b8771980-be7b-11eb-989a-50c7dc70ffe7.png)
1. Click "Local windows debugger".
    ![image](https://user-images.githubusercontent.com/1730073/119689721-ab5a2a80-be7b-11eb-85ed-786330592c46.png)
1. Check local variables.
    ![image](https://user-images.githubusercontent.com/1730073/119689667-9f6e6880-be7b-11eb-9d8e-124070b7ff1c.png)
1. Click "Continue".
    ![image](https://user-images.githubusercontent.com/1730073/119689609-92517980-be7b-11eb-901e-2fb0238f469d.png)
1. If nothing is displayed, click “Break All”. 
    ![image](https://user-images.githubusercontent.com/1730073/119689400-6635f880-be7b-11eb-9bb4-fa686cafd50e.png)
1. Check Threads and Call stack.
    ![image](https://user-images.githubusercontent.com/1730073/119689553-882f7b00-be7b-11eb-955b-cd25b0d7b790.png)

**Comment by Felixmaker** [comment]
Thank you! I switch to v142_xp and build the program successful.  I checked my project, maybe when I test the program in XP, I forgot to move the .xml file at the same time, so the window do not show. 

👍 I have tested the program, it just works fine! 👍 

![run in xp](https://user-images.githubusercontent.com/7445790/119759457-df186d00-beda-11eb-9afa-63ed54147332.png)


PS: If we want to support XP, we need to disable the touch support first in config.h (maybe XP do not support touch!) and then use v142_xp compilation toolchain to build.
